e2e/src/helper.ts

```diff
@@ -30,41 +31,55 @@ type voidFunc = () => void;
 const PROJECTS_PATH = path.resolve(__dirname, '../projects');
 const DOWNLOADS_PATH = path.resolve(__dirname, '../.downloads');
 export const TEST_FIXTURE_PATH = path.resolve(PROJECTS_PATH, 'test-fixture');
+export const POSTGRES_PATH = path.resolve(PROJECTS_PATH, 'postgres');
 export const PREVIEW_URI = 'query-preview:Preview?dbt-language-server';
 
 export const MAX_VSCODE_INTEGER = 2147483647;
 export const MAX_RANGE = new Range(0, 0, MAX_VSCODE_INTEGER, MAX_VSCODE_INTEGER);
 export const MIN_RANGE = new Range(0, 0, 0, 0);
 
+export const LS_MORE_THAN_OPEN_DEBOUNCE = 1005;
+
 workspace.onDidChangeTextDocument(onDidChangeTextDocument);
+
 window.onDidChangeActiveTextEditor(e => {
   console.log(`Active document changed: ${e?.document.uri.toString() ?? 'undefined'}`);
 });
 
 let previewPromiseResolve: voidFunc | undefined;
 let documentPromiseResolve: voidFunc | undefined;
 
+let extensionApi: ExtensionApi | undefined = undefined;
+const languageServerReady = new Map<string, DeferredResult<void>>();
+
 let tempModelIndex = 0;
 
-export async function activateAndWait(docUri: Uri): Promise<void> {
-  // The extensionId is `publisher.name` from package.json
-  const ext = extensions.getExtension('Fivetran.dbt-language-server');
-  if (!ext) {
-    throw new Error('Fivetran.dbt-language-server not found');
-  }
+export async function openTextDocument(docUri: Uri): Promise<void> {
+  await workspace.openTextDocument(docUri);
+  editor = await window.showTextDocument(docUri);
+}
 
-  const existingEditor = window.visibleTextEditors.find(e => e.document.uri.path === docUri.path);
+export async function activateAndWait(docUri: Uri): Promise<void> {
+  const existingEditor = findExistingEditor(docUri);
   const doNotWaitChanges = existingEditor && existingEditor.document.getText() === window.activeTextEditor?.document.getText() && getPreviewEditor();
   const activateFinished = doNotWaitChanges ? Promise.resolve() : createChangePromise('preview');
 
-  await ext.activate();
-
   doc = await workspace.openTextDocument(docUri);
   editor = await window.showTextDocument(doc);
   await showPreview();
   await activateFinished;
 }
 
+export async function activateAndWaitManifestParsed(docUri: Uri, projectFolderName: string): Promise<void> {
+  doc = await workspace.openTextDocument(docUri);
+  editor = await window.showTextDocument(doc);
+  await Promise.all([sleep(LS_MORE_THAN_OPEN_DEBOUNCE), waitForManifestParsed(projectFolderName)]);
```

Comment:
Probably we don't need to wait `LS_MORE_THAN_OPEN_DEBOUNCE` if editor is already open. We can add some condition using `findExistingEditor` function.